spec/index.html

```diff
@@ -8941,13 +8953,120 @@ <h5>Collect <code>FILTER</code> Elements</h5>
+              <p><b>prepare(EXISTS{P})</b></p>
+
+              <div class="ednote">
+                <p>@@ Scoping rule already done (grammar note).<br/>
+                  Prepare FILTER for exists. 
+                </p>
+                <ul>
+                  <li>A1 = translate(P) without applying simplification</li>
+                  <li>A2 = A1 rewritten to include access current binding</li>
+                  <li>A3 = A2 Remapped hidden-scope variables (if we do that)</li>
+                </ul>
+
+                <div class="defn">
+                  <div>
+                    <b>Definition: Access to the current binding</b>
+                    <p>
+                      During <a href="#sparqlQuery">translation to the SPARQL algebra</a>
+                    </p>
+                  <pre>
+Replace each occurence of `Y` in X where `Y` is one of
+          <a href="#sparqlTranslateBasicGraphPatterns">Basic Graph Pattern</a>,
+          <a href="#sparqlTranslatePathExpressions">Property Path Expression</a>,
+          <a href="#sparqlTranslateGraphPatterns">`Graph(Var, pattern)`</a>,
+          <a href="#https://www.w3.org/TR/sparql12-query/#sparqlTranslateGraphPatterns">Inline Data</a>
+          with `join(Y, BindingInScope())`.</pre>
+                  </div>
+                  <div class="note">
+                    c.f. section <a href="#sparqlTranslateGraphPatterns">Translate Graph Patterns</a>
+                    where an empty basic graph pattern start any 
+                    <a href="#rGroupGraphPattern">GroupGraphPattern</a>.
+                    It happens before the <a href="#sparqlSimplification">simplification step</a>.
+                  </div>
+                </div>
+            
+                <div class="ednote" id="note-BindingInScope">
+                  <p>
+                    One way to provide `BindingInScope` is to change `eval` to also have the current row
+                    as an argument or a "null" token. . Normally this the "null" token. This would be set in
+                    eval-filter.
+                  </p>
+                  <p>
+                    Another way is to have a global (to the execution) variable. In eval-exists, the
+                    old value is recorded, the global set to the new value, and reset on exit - this forms a
+                    stack for nested EXISTS.
+                  </p>
+                  <p>
+                    Modifing `eval` is better prepartion for other correlted operations.<br/>
+                    A current-row stack in EXISTS isless intrusive.
+                  </p>
+                </div>
+                
+                <div class="defn">
+                  <b>Definition: <span id="defn_projmap">Projection Expression Variable Remapping</span></b>
+                  <p>
+                    For a projection algebra operation #sparqlProjection `Project(A, PV)` acting on algreg express `A` and with set of variables `PV`, define
```

Comment:
My suggested `algebra regular expression` might should be `algebraic regular expression`.
```suggestion
                    For a projection algebra operation #sparqlProjection `Project(A, PV)` acting on algebra regular expression `A` with set of variables `PV`, define
```